Sources/Overture/Zurry.swift

```diff
@@ -6,3 +6,19 @@
 public func zurry<A>(_ function: @escaping () throws -> A) rethrows -> A {
   return try function()
 }
+
+/// Wraps a value in a function.
+///
+/// - Parameter value: A value.
+/// - Returns: A wrapper function.
+public func unzurry<A>(_ value: @autoclosure @escaping () -> A) -> () -> A {
```

Comment:
What do you think of making unzurry lazy? I wonder if that'd fix the compiler issue you had earlier.